lib/ansible/modules/cloud/amazon/ec2_vpc_igw.py

```diff
@@ -67,63 +74,110 @@
 
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.six import string_types
 
 
 class AnsibleIGWException(Exception):
     pass
 
 
-def ensure_igw_absent(vpc_conn, vpc_id, check_mode):
-    igws = vpc_conn.get_all_internet_gateways(
-        filters={'attachment.vpc-id': vpc_id})
+def get_igw_info(igw):
+    return {'id': igw.id,
+            'tags': igw.tags,
+            'vpc_id': igw.vpc_id
+            }
 
-    if not igws:
-        return {'changed': False}
 
-    if check_mode:
-        return {'changed': True}
+def get_resource_tags(vpc_conn, resource_id):
+    return dict((t.name, t.value) for t in
+                vpc_conn.get_all_tags(filters={'resource-id': resource_id}))
 
-    for igw in igws:
-        try:
-            vpc_conn.detach_internet_gateway(igw.id, vpc_id)
-            vpc_conn.delete_internet_gateway(igw.id)
-        except EC2ResponseError as e:
-            raise AnsibleIGWException(
-                'Unable to delete Internet Gateway, error: {0}'.format(e))
 
-    return {'changed': True}
+def ensure_tags(vpc_conn, resource_id, tags, add_only, check_mode):
```

Comment:
Should probably use `compare_aws_tags` for future reference but not a blocker here as it is only recently available